### PR TITLE
[AWS] Add agent.id field to be set as dimension for TSDB migration

### PR DESCRIPTION
**Pull request by constanca-m**
## What does this PR do?

The field `agent.id` needs to be set as dimension for TSDB migration. Reasoning for this can be found in the thread following the comment https://github.com/elastic/integrations/issues/5193#issuecomment-1536090359

Note: `agent.id` hasn't been added to all data streams yet, just to the ones that were migrated to TSDB already.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/aws/manifest.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 format_version: 1.0.0
 name: aws
 title: AWS
-version: 1.37.3
+version: 1.38.0
```
[the remark this answers is not in the document]
Comment (constanca-m) [reply packages/aws/manifest.yml:4]: Changes on 1.37 are not related to TSDB, that is why I bumped to 1.30

**Comment by constanca-m** [comment]
> Please confirm that you have verified the number of dimensions are within 16 , after making this change, for all datastreams?

Yes @agithomas:
DynamoDB: 11
EBS: 4
ECS: 5
ELB: 7
Kinesis: 4
Lambda: 6
Natgateway: 4
Redshift: 11
Usage: 7

**Review by agithomas — APPROVED** [review]
LGTM!

**Review by tommyers-elastic — APPROVED** [review]
LGTM

**Pull request merged**